daemon-tests/tests/happy_path.rs

```diff
@@ -427,6 +427,42 @@ async fn maker_rejects_rollover_after_commit_finality() {
     wait_next_state!(order_id, maker, taker, CfdState::OpenCommitted);
 }
 
+#[tokio::test]
+async fn maker_accepts_rollover_after_commit_finality() {
```

Comment:
This test showcases that something is wrong at the moment.
Are you planning to fix this bug as part of this PR?

I think it would be better if we write a failing test and then fix this issue. - Or ignore the test for now. 
I found it a bit confusing to have a test that tests that something is a bug.

Reply:
This test actually shows that I at least partially fixed a problem in this PR! (it was failing before the changes).
I am planning to tackle the remaining issue today.